draft-ounsworth-cfrg-kem-combiners.mkd

```diff
@@ -280,13 +318,16 @@ This is a non-comprehensive list, further information can be found in paragraph
 
 # Practical instantiations {#sec-instantiation}
 
-The KDF MUST be instantiated with one of the following Keccak-based option.
-Each instance defines a function to be used as `KDF`, a hash `H` function to optionally derive the `k_i`, and a `counter`.
+The KDF must be instantiated with cryptographically-secure choices for `KDF` and `H`. The following are RECOMMENDED Keccak-based instatiations, but other choices MAY be made depending on availability of cryptographic primitives, protocol design needs, and especially to allow for future cryptographic agility.
+
+Each instance defines a function to be used as `KDF`, a hash `H` function to derive the `k_i`, and optionally a `counter`.
 
 1. `KDF = SHA3-256` and `H = SHA3-256`, with `hashSize = 256 bit`.
-2. `KDF = SHA3-512` and `H = SHA3-512`, with `hashSize = 512 bit`.
-3. `KDF = KMAC128` and `H = SHA3-256`, with `hashSize = 128 bit`.
-4. `KDF = KMAC256` and `H = SHA3-512`, with `hashSize = 256 bit`.
+1. `KDF = SHA3-512` and `H = SHA3-512`, with `hashSize = 512 bit`.
+1. `KDF = SHAKE128` and `H = SHA3-256`, with `hashSize` at least 128 bit.
+1. `KDF = SHAKE256` and `H = SHA3-512`, with `hashSize` at least 256 bit.
+1. `KDF = KMAC128` and `H = SHA3-256`, with `hashSize = 128 bit`.
+1. `KDF = KMAC256` and `H = SHA3-512`, with `hashSize = 256 bit`.
 
 ## Hash-and-counter based construction
```

Comment:
Question: What is the intention here? To be compatible to HKDF_Expand or NIST SP-800 56Cr2 or define something new?

Depending on what is intended here I would check for compliance.

Reply:
I would also like to remove the counter and streamline the construction, but is it feasible to then still call it inline with `NIST SP-800 56Cr2`?